sstables/mc/parquet_writer.hh

```diff
@@ -0,0 +1,973 @@
+                break;
+            }
+            case kind::time: {
+                auto& w = _writer->column<map_physical_type(kind::time)>(writer_id);
+                auto typed_value = value_cast<int64_t>(dv);
+                w.put(def, rep, typed_value);
+                break;
+            }
+            case kind::timestamp: {
+                auto& w = _writer->column<map_physical_type(kind::timestamp)>(writer_id);
+                auto typed_value = value_cast<db_clock::time_point>(dv);
+                int64_t millis = typed_value.time_since_epoch().count();
+                w.put(def, rep, millis);
+                break;
+            }
+            case kind::utf8:{
+                auto& w = _writer->column<map_physical_type(kind::utf8)>(writer_id);
+                auto typed_value = value_cast<sstring>(dv);
+                w.put(def, rep, sstring_to_bytes_view(typed_value));
+                break;
+            }
+            }
+        } catch (const std::bad_cast& e) {
+            parquet_logger.error("Deserialization error in table {}.{}, column {}, type {},  ordinal_id {}, parquet column {}",
+                    _pws.scylla_sch->ks_name(), _pws.scylla_sch->cf_name(), col_def.name_as_text(), col_def.type->name(), ordinal_id, writer_id);
+            write_cell_value_empty(ordinal_id);
+        }
+    }
+    void write_cell_value_empty(int id) {
+        if (std::holds_alternative<UNKNOWN>(_pws.cell_mappings[id].pq_type)) {
+            return;
+        }
+        std::visit(parquet4seastar::overloaded {
+            [&] (const auto& lt) {
+                constexpr parquet4seastar::format::Type::type pt = lt.physical_type;
+                auto& w = _writer->column<pt>(_pws.cell_mappings[id].value);
+                using input_type = typename std::remove_reference_t<decltype(w)>::input_type;
+                w.put(1, rep(), input_type{});
+            },
+            [&] (const parquet4seastar::logical_type::INT96&) {
+                // unreachable
+            },
+        }, _pws.cell_mappings[id].pq_type);
+    }
+    void write_cell_empty(int id) {
+        write_cell_value_empty(id);
+        write_cell_metadata<parts::ROW_CELLS_REGULAR_X_FLAGS>(id, 1, rep(), 0);
+        write_cell_metadata<parts::ROW_CELLS_REGULAR_X_DT>(id, 1, rep(), 0);
+        write_cell_metadata<parts::ROW_CELLS_REGULAR_X_DLDT>(id, 1, rep(), 0);
+        write_cell_metadata<parts::ROW_CELLS_REGULAR_X_DTTL>(id, 1, rep(), 0);
+    }
+    void write_row_fill() {
+        for (const auto& col_def : _pws.scylla_sch->regular_columns()) {
+            int id = (int)col_def.ordinal_id;
+            if (std::holds_alternative<UNKNOWN>(_pws.cell_mappings[id].pq_type)) {
+                continue;
+            }
+            if (_cells_written[id]) {
+                _cells_written[id] = false;
+                continue;
+            }
+            write_cell_empty(id);
+        }
+    }
+    void write_srow_fill() {
```

Comment:
let's expand all this names. `write_static_row_fill` is still rather short, and srow is not standard nomenclature